### ISSUE TITLE
Nomic 1.5 embedding model load error

### ISSUE THREAD

**Issue by el95149** [lmstudio-ai/lmstudio-bug-tracker#995]
**Which version of LM Studio?**
LM Studio 0.3.25 (Build 2)

**Which operating system?**
Kubuntu 25.04

**What is the bug?**
Trying to load the bundled Nomic 1.5 embedding model consistently fails, as of a couple of builds ago (IIRC).
Happens when using default model load parameters, as well as non-default ones (e.g. 'Flash Attention').
Opening the file path mentioned in the error log shows the model file with a size of 133B.

**Screenshots**

<img width="847" height="953" alt="Image" src="https://github.com/user-attachments/assets/31ed0207-5d6a-47c4-88a8-b55de8e91772" />

<img width="608" height="338" alt="Image" src="https://github.com/user-attachments/assets/4d3dfa98-6b96-424d-b46c-21ad9d09cae7" />

<img width="1254" height="218" alt="Image" src="https://github.com/user-attachments/assets/4f3b7e6f-eaf6-42bc-9c2d-4afe2554448d" />

**Logs**
```
2025-09-10 08:51:07 [DEBUG]
 [LM Studio] GPU Configuration:
  Strategy: evenly
  Priority: []
  Disabled GPUs: []
  Limit weight offload to dedicated GPU Memory: OFF
  Offload KV Cache to GPU: ON
2025-09-10 08:51:07 [DEBUG]
 [LM Studio] Live GPU memory info (source 'LMS Core'):
  GPU 0: NVIDIA GeForce RTX 4080 (Used: 3.77 GB, Total: 17.17 GB, Free: 13.40 GB)
2025-09-10 08:51:07 [DEBUG]
 [LM Studio] Model load size estimate with raw num offload layers '0' and context length '2048':
  Model: 0
  Context: 162.53 MB
  Total: 162.53 MB
2025-09-10 08:51:07 [DEBUG]
 [LM Studio] Strict GPU VRAM cap is OFF: GPU offload layers will not be checked for adjustment
2025-09-10 08:51:07 [DEBUG]
 [LM Studio] Resolved GPU config options:
  Num Offload Layers: 0
  Num CPU Expert Layers: 0
  Main GPU: 0
  Tensor Split: [0]
  Disabled GPUs: []
2025-09-10 08:51:07 [DEBUG]
 ggml_cuda_init: GGML_CUDA_FORCE_MMQ:    no
ggml_cuda_init: GGML_CUDA_FORCE_CUBLAS: no
ggml_cuda_init: found 1 CUDA devices:
2025-09-10 08:51:07 [DEBUG]
   Device 0: NVIDIA GeForce RTX 4080, compute capability 8.9, VMM: yes
2025-09-10 08:51:07 [DEBUG]
 CUDA : ARCHS = 500,610,750,800 | USE_GRAPHS = 1 | PEER_MAX_BATCH_SIZE = 128 | CPU : SSE3 = 1 | SSSE3 = 1 | AVX = 1 | AVX2 = 1 | F16C = 1 | FMA = 1 | BMI2 = 1 | LLAMAFILE = 1 | OPENMP = 1 | REPACK = 1 |
2025-09-10 08:51:07 [DEBUG]
 llama_model_load_from_file_impl: using device CUDA0 (NVIDIA GeForce RTX 4080) - 12779 MiB free
gguf_init_from_file_impl: invalid magic characters: 'vers', expected 'GGUF'
2025-09-10 08:51:07 [DEBUG]
 llama_model_load: error loading model: llama_model_loader: failed to load model from /tmp/.mount_LM-StufzUqje/resources/app/.webpack/bin/bundled-models/nomic-ai/nomic-embed-text-v1.5-GGUF/nomic-embed-text-v1.5.Q4_K_M.gguf
llama_model_load_from_file_impl: failed to load model
common_init_from_params: failed to load model '/tmp/.mount_LM-StufzUqje/resources/app/.webpack/bin/bundled-models/nomic-ai/nomic-embed-text-v1.5-GGUF/nomic-embed-text-v1.5.Q4_K_M.gguf', try reducing --n-gpu-layers if you're running out of VRAM
lmstudio-llama-cpp: failed to load model. Error: error loading model: llama_model_loader: failed to load model from /tmp/.mount_LM-StufzUqje/resources/app/.webpack/bin/bundled-models/nomic-ai/nomic-embed-text-v1.5-GGUF/nomic-embed-text-v1.5.Q4_K_M.gguf
```

**To Reproduce**
1. Startup LM Studio
2. Click on 'Select a model to load'
3. Choose 'Nomic Embedded Text v1.5'
4. See error.

**Comment by justinbadal** [comment]
I'm experiencing the same error.

**Comment by undeference** [comment]
At least the AppImage was shipped with a reference and not the actual model. It does not automatically fetch it (nor does it seem to load it from the normal location, so it wouldn't work anyway). The solution is to extract the contents of the AppImage (`./LM-Studio*.AppImage --appimage-extract`) and save the model from <https://huggingface.co/nomic-ai/nomic-embed-text-v1.5-GGUF/resolve/main/nomic-embed-text-v1.5.Q4_K_M.gguf> to `squashfs-root/resources/app/.webpack/bin/bundled-models/nomic-ai/nomic-embed-text-v1.5-GGUF/nomic-embed-text-v1.5.Q4_K_M.gguf`.

**Comment by el95149** [comment]
@undeference thanks for the tip! Just tried it, but I'm not sure on how I should run the extracted app. Trying both:

```bash
# from either inside `squashfas-root` or one dir above it
AppRun --no-sandbox 
#results in:
AppRun: line 45: /lm-studio: No such file or directory
# and
lm-studio --no-sandbox
# results in the app starting, but all frozen up, and me having to kill its process
```

I'm not very well versed in AppImages, so any help will be welcome!

**Comment by undeference** [comment]
> # from either inside `squashfas-root` or one dir above it
> AppRun --no-sandbox 
> #results in:
> AppRun: line 45: /lm-studio: No such file or directory

Same. Try without `--no-sandbox`?

**Comment by el95149** [comment]
@undeference still no-go, `--no-sandbox` seems to be a staple for AppImages on more recent (K)Ubuntu versions noawadays.

```bash
[186977:0913/211437.611150:FATAL:sandbox/linux/suid/client/setuid_sandbox_host.cc:169] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /home/<readacted>/Applications/squashfs-root/chrome-sandbox is owned by root and has mode 4755.
Trace/breakpoint trap (core dumped)
```
It's OK, they'll fix it at some point, I guess.

**Comment by undeference** [comment]
> [@undeference](https://github.com/undeference) still no-go, `--no-sandbox` seems to be a staple for AppImages on more recent (K)Ubuntu versions noawadays.
> 
> [186977:0913/211437.611150:FATAL:sandbox/linux/suid/client/setuid_sandbox_host.cc:169] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /home/<readacted>/Applications/squashfs-root/chrome-sandbox is owned by root and has mode 4755.
> Trace/breakpoint trap (core dumped)
> 
> It's OK, they'll fix it at some point, I guess.

It works for me after rebuilding with appimagetool-x86_64.AppImage from <https://github.com/AppImage/appimagetool/releases/tag/continuous> (after fixing the lm-studio.desktop file).

I considered uploading my rebuild somewhere but I'm not sure the license allows that. Also, you shouldn't trust me.

**Comment by el95149** [comment]
@undeference I'll try it on my own, don't worry. Thanks for all the tips!

**Comment by el95149** [comment]
Reporting back that the bug has been fixed on v0.3.26 (Build 6), at least for me. If others corroborate, I'll close the ticket.

**Comment by undeference** [comment]
LM Studio 0.3.26 (Build 6) works for me.

**Comment by el95149** [comment]
Closing, since it seems to be fixed.

**Comment by freuskadi** [comment]
Hello,
For me, RAG is still not working in LM Studio 0.3.26.

System : Linux Mint (no GPU, 64 GB RAM)
Tested with : Qwen3, gpt-oss, llama3.1

What I tried : 
- Adding extra instructions in the prompt
- Asking in French and in English
- Attaching both short and simple text files

Observed behavior : 
- nomic-embed-text-v1.5 does not appear in the interface.
- When I try to download it, LM Studio tells me it is already loaded.
- I can see my file is correctly attached in the UI, but in the developer logs there is nothing related to /nomic-embed-text-v1.5 or embeddings.
- In “Thinking” mode, the model explicitly says : “If no files attached or no RAG data in the context, respond ‘I don’t look attached file’.” → This shows that no RAG data is ever injected into the context.

Conclusion : 
So far, I have never been able to get RAG or MCP working in LM Studio on Linux with my setup. It seems that the embedding model is bundled inside the AppImage, but the RAG pipeline never actually triggers.

Developer logs : 
2025-09-16 21:18:13 [DEBUG]
 Prompt successfully formatted with Harmony.
2025-09-16 21:18:13 [DEBUG]
 Prompt successfully formatted with Harmony.
2025-09-16 21:18:13 [DEBUG]
 Sampling params:	repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 40, top_p = 0.800, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.800
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
Sampling: 
logits -> logit-bias -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
Generate: n_ctx = 2016, n_batch = 512, n_predict = -1, n_keep = 0
2025-09-16 21:18:13 [DEBUG]
 Cache reuse summary: 40/293 of prompt (13.6519%), 40 prefix, 0 non-prefix
Total prompt tokens: 293
Prompt tokens to decode: 253
BeginProcessingPrompt
2025-09-16 21:18:23 [DEBUG]
 FinishedProcessingPrompt. Progress: 100

**Comment by undeference** [comment]
I have the same thing but nomic-embed is loaded and context injection works so it's a different bug.
